### ISSUE TITLE
git module always fails on update if local has modification

### ISSUE THREAD

**Issue by wangjild** [ansible/ansible-modules-core#5292]
##### ISSUE TYPE

- Bug Report
##### COMPONENT NAME

git
##### ANSIBLE VERSION

```
2.2.0
```
##### OS / ENVIRONMENT

debian:8.0 jessie
##### SUMMARY

If local git repository has modification, an update attempt of always fails with Local modifications exist, even if force=yes was given.
##### STEPS TO REPRODUCE

```
  tasks:
    - name: update project dependency
      git: dest={{item.location|quote}} repo={{item.scm_url|quote}} version={{item.scm_revision|quote}} force=yes refspec={{item.scm_refspec}} accept_hostkey=yes
      with_items: "{{ deps }}"
```

##### ACTUAL RESULTS

```
failed: [127.0.0.1] => {
    "failed": true,
    "invocation": {
        "module_name": "git"
    },
    "item": {
        "location": "/opt/tiger/neihan/conf",
        "name": "neihan/conf",
        "scm_refspec": "refs/heads/master",
        "scm_revision": "master",
        "scm_url": "ssh://*********/neihan/conf"
    },
    "module_stderr": "Shared connection to 127.0.0.1 closed.\r\n",
    "module_stdout": "Traceback (most recent call last):\r\n  File \"/tmp/ansible_U00Fwd/ansible_module_git.py\", line 1023, in <module>\r\n    main()\r\n  File \"/tmp/ansible_U00Fwd/ansible_module_git.py\", line 974, in main\r\n    result.update(changed=True, after=remote_head, msg='Local modifications exist')\r\nUnboundLocalError: local variable 'remote_head' referenced before assignment\r\n",
    "msg": "MODULE FAILURE"
}
```

**Comment by ansibot** [comment]
@ansible ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mscherer** [comment]
Isn't it a duplicate of https://github.com/ansible/ansible-modules-core/pull/5260 ?

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29574
